packages/sdk-react/src/components/receivables/InvoicesTable/InvoicesTable.tsx

```diff
@@ -36,15 +36,56 @@ import { GridSortDirection } from '@mui/x-data-grid/models/gridSortModel';
 import { Filters } from '../Filters';
 import { useReceivablesFilters } from '../Filters/useReceivablesFilters';
 import { InvoiceCounterpartName } from '../InvoiceCounterpartName';
+import {
+  type InvoicesTableRowAction,
+  type InvoicesTableRowActionSchema,
+  useInvoiceRowActionMenuCell,
+} from './useInvoiceRowActionMenuCell';
 
-type InvoicesTableProps = {
+export interface InvoicesTableProps {
   /**
    * The event handler for a row click.
    *
    * @param id - The identifier of the clicked row, a string.
    */
   onRowClick?: (id: string) => void;
-};
+
+  /**
+   * The event handler for an invoice row action.
+   * If not provided, the action menu will not be displayed.
```

Comment:
As I understand we have 2 options:
1. User doesn't provide `onRowActionClick` & `rowActions`
2. User provides `onRowActionClick` & `rowActions`

Because if the user provides `onRowActionClick` we can't show row actions. And vice-versa if the customer provides `rowActions` but not `onRowActionClick` we can't show UI with no callbacks.

So my idea is to have a united type, as an example:

```ts
interface InvoicesTableInternalProps {
  onRowClick?: (id: string) => void;
}

interface InvoicesTableRowActionsProps extends InvoicesTableInternalProps {
  // Make it required!
  onRowActionClick: ...
  
  rowActions?: ...
}

type InvoicesTableProps = InvoicesTableInternalProps | InvoicesTableRowActionsProps
```

In that case, if the customer would like to provide `rowActions` they have to specify the `onRowActionClick` callback. If they don't so in that case everything will be the same.

[We already use it for](https://github.com/team-monite/monite-sdk/blob/29d69da72605a506ac29437592d606829158bf01/packages/sdk-react/src/components/receivables/InvoiceDetails/InvoiceDetails.types.ts#L100) `InvoiceDetails` component as an example when we specify `create` or `update` scenarios

So WDYT?

Reply:
Thank you for the great idea! Refactored [in another way](https://github.com/team-monite/monite-sdk/blob/abe0d7ebbdb4dc4435fef87fd6b55368e34aa4d6/packages/sdk-react/src/components/receivables/InvoicesTable/useInvoiceRowActionMenuCell.tsx#L18-L54).